src/main/java/arcaratus/bloodarsenal/compat/guideapi/GuideBloodArsenal.java

```diff
@@ -24,38 +25,23 @@
 @GuideBook(priority = EventPriority.HIGHEST)
 public class GuideBloodArsenal implements IGuideBook
 {
-    public static final Book GUIDE_BOOK = new Book();
-
     @Nullable
     @Override
     public Book buildBook()
     {
-        GUIDE_BOOK.setTitle("guide.bloodarsenal.title");
-        GUIDE_BOOK.setDisplayName("guide.bloodarsenal.display");
-        GUIDE_BOOK.setWelcomeMessage("guide.bloodarsenal.welcome");
-        GUIDE_BOOK.setAuthor("guide.bloodarsenal.author");
-        GUIDE_BOOK.setRegistryName(new ResourceLocation(BloodArsenal.MOD_ID, "guide"));
-        GUIDE_BOOK.setColor(new Color(119, 0, 0));
-
-        return GUIDE_BOOK;
-    }
-
-//    @Override
-//    public void handleModel(ItemStack bookStack)
-//    {
-//        GuideAPI.setModel(GUIDE_BOOK);
-//    }
-
-    @Override
-    public void handlePost(ItemStack bookStack)
-    {
-        String keyBase = "guide.bloodarsenal.category.";
-        GUIDE_BOOK.addCategory(new CategoryItemStack(CategoryLifebringer.buildCategory(), keyBase + "lifebringer", EnumBaseTypes.BLOOD_INFUSED_IRON_INGOT.getStack()));
+        BookBinder binder = new BookBinder(new ResourceLocation(BloodArsenal.MOD_ID, "guide"));
+        binder.setGuideTitle("guide.bloodarsenal.title");
+        binder.setItemName("guide.bloodarsenal.display");
+        binder.setHeader("guide.bloodarsenal.welcome");
+        binder.setAuthor("guide.bloodarsenal.author");
+        binder.setColor(new Color(119, 0, 0));
+        binder.addCategory(new CategoryItemStack(CategoryLifebringer.buildCategory(), "guide.bloodarsenal.category.lifebringer", EnumBaseTypes.BLOOD_INFUSED_IRON_INGOT.getStack()));
+        return binder.build();
     }
 
     @Override
     public IRecipe getRecipe(@Nonnull ItemStack bookStack)
     {
-        return new ShapelessOreRecipe(new ResourceLocation(BloodArsenal.MOD_ID, "guide"), GuideAPI.getStackFromBook(GUIDE_BOOK), Items.BOOK, Items.FLINT_AND_STEEL, RegistrarBloodMagicItems.BLOOD_ORB, FluidUtil.getFilledBucket(new FluidStack(BlockLifeEssence.getLifeEssence(), 0))).setRegistryName("bloodarsenal_guide");
+        return new ShapelessOreRecipe(new ResourceLocation(BloodArsenal.MOD_ID, "guide"), GuideAPI.getStackFromBook(buildBook()), Items.BOOK, Items.FLINT_AND_STEEL, RegistrarBloodMagicItems.BLOOD_ORB, FluidUtil.getFilledBucket(new FluidStack(BlockLifeEssence.getLifeEssence(), 0))).setRegistryName("bloodarsenal_guide");
```

Comment:
No no no, that's not how this works. `buildBook` is called by GuideAPI itself. Just as with the previous implementation, an instance should be stored as [mentioned in the javadoc](https://github.com/TeamAmeriFrance/Guide-API/blob/1.12/src/main/java/amerifrance/guideapi/api/IGuideBook.java#L15-L16).

Reply:
Thanks, I thought it called it itself, I just skipped over the line saying that a reference should be kept. I made a change, should that be fine now? Or should I add a check to make sure that it is initialized so that it doesn't have a chance of passing null to the recipe.

Reply:
This should be fine. `buildBook` is called during `Register<Biome>` ([yes I know that sounds weird](https://github.com/TeamAmeriFrance/Guide-API/blob/1.12/src/main/java/amerifrance/guideapi/RegistrarGuideAPI.java#L28-L34)) while `getRecipe()` is called during `Register<IRecipe>`.